Docs: Update section on plugin dev for explore (#31905)
grafana/grafana: docs/sources/developers/plugins/add-support-for-explore-queries.md

   };
   ```

## Selecting preferred visualisation

> Explore should be default select a reasonable visualisation for your data so users do not have to tweak and play with the visualisations and just focus on querying. This usually works fairly well and Explore can figure out whether the returned data is time series data or logs or something else.

Reviewer: ```suggestion
Explore should be the default. Select a reasonable visualization for your data so users do not have to tweak and play with the visualizations and just focus on querying. This usually works well and Explore can figure out whether the returned data is time series data or logs or something else.
```

Author: Ah that should have been `Explore should by default select` :D

> Occasionally this may not work or you really want to show some data in a specific visualisation. In that cases you can add a hint to your returned data frame, setting the preferredVisualisationType meta attribute.

Reviewer: ```suggestion
Occasionally this may not work or you may want to show some data in a specific visualization. In that case, you can add a hint to your returned data frame, setting the `preferredVisualisationType` meta attribute.
```

Reviewer 2: I recommend replacing "may" with "might." 

Per the [Microsoft style guide](https://docs.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/c/can-may): "When ability is what you need to express, it's OK to use can to describe actions or tasks that the reader or program is able to do. Use _might_ to express possibility. Don't use _may_, which might be interpreted as providing permission."

Author: hmm in the case of `or you may want to' -> `or you might want to` does that make sense? Seems like a strange way to say it.

Reviewer 3: For consistency, use _might_ for probability and _may_ for permission. +1 for @oddlittlebird’s comment.

The reason why this might **sound** odd, is that it is largely what is used in spoken language versus written language.

> You can construct data frame with specific metadata like this:

Reviewer 3: ```suggestion
You can construct a data frame with specific metadata:
```

```
const firstResult = new MutableDataFrame({
    fields: [...],
    meta: {
        preferredVisualisationType: 'logs',
    },
});
```

> See [PreferredVisualisationType](https://grafana.com/docs/grafana/latest/packages_api/data/preferredvisualisationtype/) for possible options. 

Reviewer 3: ```suggestion
For possible options, refer to [PreferredVisualisationType](https://grafana.com/docs/grafana/latest/packages_api/data/preferredvisualisationtype/). 
```
Keep consistent with "refer to" verbiage instead of "see".

## Selecting preferred visualisation

Explore should by default select a reasonable visualization for your data so users do not have to tweak and play with the visualizations and just focus on querying. This usually works fairly well and Explore can figure out whether the returned data is time series data or logs or something else.

> Occasionally this might not work or you might want to show some data in a specific visualization. In that case, you can add a hint to your returned data frame, setting the `preferredVisualisationType` meta attribute.

Reviewer: I see you have not accepted the suggestion. Not a blocker, although the suggestion would improve doc content by removing extra words and adding clarity.

> Occasionally this might not work or you might want to show some data in a specific visualization. In that case, you can add a hint to your returned data frame, setting the `preferredVisualisationType` meta attribute.

Reviewer 2: Three of three documentarians agree with this suggestion.

> Occasionally this might not work or you might want to show some data in a specific visualization. In that case, you can add a hint to your returned data frame, setting the `preferredVisualisationType` meta attribute.

Author: @achatterjee-grafana not sure which one do you mean. I updated the section manually and pushed the changes because I also merged in the suggestion from @oddlittlebird about may -> might.
